crates/cxx-qt-lib/src/core/qflags/qflag.rs

```diff
@@ -0,0 +1,106 @@
+// SPDX-FileCopyrightText: 2025 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Joshua Booth <joshua.n.booth@gmail.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Not};
+
+use cxx::ExternType;
+
+pub trait QFlagRepr: Sized {
```

Comment:
Please mark this trait as private, as it should only be implemented here, and not by downstream consumers.

Reply:
I think it's useful for users to know which types implement `QFlagRepr`, so I've sealed the trait to prevent downstream implementations.

Reply:
Cool, that works as well :+1: